sql/wsrep_sst.cc

```diff
@@ -273,6 +288,140 @@ struct sst_thread_arg {
   }
 };
 
+struct sst_logger_thread_arg {
+
+  /* This is the READ end of the stderr pipe */
+  FILE *  err_pipe;
+
+  sst_logger_thread_arg(FILE *ferr)
+    : err_pipe(ferr) {
+    }
+
+  ~sst_logger_thread_arg() {
+    if (err_pipe)
+      fclose(err_pipe);
+    err_pipe = NULL;
+  }
+};
+
+static enum loglevel string_to_loglevel(const char *s)
+{
+  if (strncmp(s, "ERR:", 4) == 0)
+    return ERROR_LEVEL;
+  else if (strncmp(s, "WRN:", 4) == 0)
+    return WARNING_LEVEL;
+  else if (strncmp(s, "INF:", 4) == 0)
+    return INFORMATION_LEVEL;
+  else if (strncmp(s, "DBG:", 4) == 0)
+    return INFORMATION_LEVEL;
+  else
+    return SYSTEM_LEVEL;
+}
+
+static void *sst_logger_thread(void *a) {
+  sst_logger_thread_arg *arg = (sst_logger_thread_arg *)a;
+
+#ifdef HAVE_PSI_INTERFACE
+  wsrep_pfs_register_thread(key_THREAD_wsrep_sst_logger);
+#endif /* HAVE_PSI_INTERFACE */
+
+  int const   out_len = 1024;
+  char        out_buf[out_len];
+
+  /* Loops over the READ end of stderr.  Only stops
+     when the pipe is in an error or EOF state.
+   */
+  while (!feof(arg->err_pipe) && !ferror(arg->err_pipe))
```

Comment:
MySQL now uses clang formatting. They ship standard clang format as part of the codebase. Do run clang formatting over codebase to be compliant with mysql. (only for wsrep-plugin can skip it for galera for now).

Reply:
Run clang-format on wsrep_sst.c[c](url)